sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -677,6 +677,103 @@ private Response<RecognizeEntitiesResultCollection> RecognizeEntitiesBatch(Multi
         #endregion
 
         #region Analyze Sentiment
+        /// <summary>
+        /// Runs a predictive model to identify the positive, negative, neutral
+        /// or mixed sentiment contained in the document, as well as a score
+        /// indicating the model's confidence in the predicted sentiment.
+        /// For a list of languages supported by this operation, see
+        /// <a href="https://docs.microsoft.com/en-us/azure/cognitive-services/text-analytics/language-support"/>.
+        /// For document length limits, maximum batch size, and supported text encoding, see
+        /// <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits"/>.
+        /// </summary>
+        /// <param name="document">The document to analyze.</param>
+        /// <param name="language">The language that the document is written in.
+        /// If unspecified, this value will be set to the default language in
+        /// <see cref="TextAnalyticsClientOptions"/> in the request sent to the
+        /// service.  If set to an empty string, the service will apply a model
+        /// where the language is explicitly set to "None".</param>
+        /// <exception cref="RequestFailedException">Service returned a non-success
+        /// status code.</exception>
+        public virtual async Task<Response<DocumentSentiment>> AnalyzeSentimentAsync(string document, string language)
```

Comment:
The purpose of this methods is to make sure that old behavior of TA still works with the new addition of `AnalyzeSentimentOptions`.
There is a conflict of guidelines here though. If a user is currently doing `AnalyzeSentiment(doc, "en") without this method, then compiler will complain that it can't decide between `AnalyzeSentiment(string, string, CancellationToken = default)` and `AnalyzeSentiment(string, string, AnalyzeSentimentOptions = default, CancellationToken = default)`.

`AnalyzeSentiment(string, string, CancellationToken = default)` => has the default because of the guideline... but then I violate the other guideline by not adding it.

I did ask KC [here](https://apiview.dev/Assemblies/Review/b196a5e53fa2441f92e3b1accefa35e7/0a7669f90b7249ddad28d589bdb1fed2#Azure.AI.TextAnalytics.TextAnalyticsClient.AnalyzeSentiment(System.String,%20System.String)) but maybe it wasn't clear enough.

I am going to change this `AnalyzeSentiment(string, string, CancellationToken = default)` to => `AnalyzeSentiment(string, string, CancellationToken)` and that will solve it all. At the end this methods is marked with EditorBrowsable Never ...

Reply:
I see. I agree that's a specific scenario. The change you proposed seems good, though.